src/main/java/codersafterdark/reskillable/base/configs/json/LockTypeJsonFactory.java

```diff
@@ -0,0 +1,40 @@
+package codersafterdark.reskillable.base.configs.json;
+
+import codersafterdark.reskillable.api.data.ItemInfo;
+import codersafterdark.reskillable.base.configs.json.parsers.CustomItemInfoJson;
+import codersafterdark.reskillable.base.configs.json.parsers.CustomItemJson;
+import codersafterdark.reskillable.base.configs.json.parsers.CustomLockJson;
+import codersafterdark.reskillable.base.configs.json.parsers.CustomNBTJson;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializer;
+import net.minecraft.item.Item;
+import net.minecraft.nbt.NBTTagCompound;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LockTypeJsonFactory {
+    private static Map<Type, JsonDeserializer> deserializerMap = new HashMap<>();
+
+    static {
+        registerDeserializer(LockJson.class, new CustomLockJson());
+        registerDeserializer(Item.class, new CustomItemJson());
+        registerDeserializer(ItemInfo.class, new CustomItemInfoJson());
+        registerDeserializer(NBTTagCompound.class, new CustomNBTJson());
+    }
+
+    public static void registerDeserializer(Type type, JsonDeserializer deserializer) {
```

Comment:
Shouldn't this be accessible through the api somehow?

Reply:
I am leaving this up to pup, as I am not sure how he wants the API to look like.

Reply:
I think this is probably fine given the JSON stuff will probably be for a larger version such as 2.0 which will allow for refactoring a bunch of stuff without worrying about breaking API things. Given a bunch of the stuff in LevelLockHandler should probably be moved from the base package to an API package. The only reason it is in LevelLockHandler to begin with is that it basically grew as I was fixing and then expanding on things making them more generic and supporting custom registrations.